config-path.js

```diff
@@ -0,0 +1,29 @@
+const path = require('path');
+const userHome = require('user-home');
+
+const env = process.env;
+const name = 'js-v8flags';
+
+function macos () {
+	const library = path.join(userHome, 'Library');
+	return path.join(library, 'Caches', name);
+}
+
+function windows () {
+	const appData = env.LOCALAPPDATA || path.join(userHome, 'AppData', 'Local');
+	return path.join(appData, name, 'Cache');
+}
+
+// https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html
+function linux () {
+	const username = path.basename(userHome);
+	return path.join(env.XDG_CACHE_HOME || path.join(userHome, '.cache'), name);
+}
+
+if (process.platform === 'darwin') {
```

Comment:
It’s not returning the function; it’s exporting it.

```js
module.exports = function (process) {
    if (process.platform === …) {
        ⋮
    }
};
```

would make it possible to pass an object in in tests without changing `process` itself, which is what causes the breakage on 0.10, and additionally clean up the require cache hack.